compiler/rustc_next_trait_solver/src/lib.rs

```diff
@@ -4,6 +4,8 @@
 //! but were uplifted in the process of making the new trait solver generic.
 //! So if you got to this crate from the old solver, it's totally normal.
 
+#![cfg_attr(not(bootstrap), allow(rustc::usage_of_type_ir_inherent))]
```

Comment:
```suggestion
#![cfg_attr(not(any(bootstrap, feature = "nightly")), allow(rustc::usage_of_type_ir_inherent))]
```

ah wait, also this

Reply:
apparently not? surprised we don't error for mentioning rustc lints but whatever 🤷‍♂️

Reply:
the tool `rustc` seems to be stable 👀; theoretically the boostrap cfg shouldn't be necessary either then by same logic

Reply:
yeah but we deny missing lints in rustc's own build config lol